crates/orderbook-commons/src/price.rs

```diff
@@ -0,0 +1,137 @@
+use crate::Order;
+use crate::ToPrimitive;
+use rust_decimal::Decimal;
+use serde::Deserialize;
+use serde::Serialize;
+use std::collections::HashMap;
+use trade::ContractSymbol;
+use trade::Direction;
+
+#[derive(Serialize, Deserialize, Default, Debug, Clone, PartialEq)]
+pub struct Price {
+    pub bid: Option<Decimal>,
+    pub ask: Option<Decimal>,
+}
+
+pub type Prices = Box<HashMap<ContractSymbol, Price>>;
+
+/// Best prices across all current orders for given ContractSymbol in the orderbook
+/// Taken orders are not included in the average
+pub fn best_current_price(current_orders: &[Order]) -> Prices {
```

Comment:
nit: It feels odd that this function is also enforcing that taken orders are not included. It feels like this should rather be a responsibility of the calling component. e.g. only fetch orders that are not taken and pass them to the `best_current_price`

Reply:
after quite a bit of thinking, I realised the taken orders by definition cannot be part of the price feed (because one *cannot* use them). 

is there any situation where this might be useful to present it as part of the price feed? this API call specifically filters the orders to get the best price, one can still access all orders if they want to.